src/__tests__/waitFor.test.tsx

```diff
@@ -263,3 +263,52 @@ test.each([false, true])(
     expect(mockFn).toHaveBeenCalledTimes(3);
   }
 );
+
+test.each([false, true])(
+  'flushes scheduled updates before returning (fakeTimers = %s)',
+  async (fakeTimers) => {
+    if (fakeTimers) {
+      jest.useFakeTimers();
+    }
+
+    function Apple({ onPress }: { onPress: (color: string) => void }) {
+      const [color, setColor] = React.useState('green');
+      const [syncedColor, setSyncedColor] = React.useState(color);
+
+      // On mount, set the color to "red" in a promise microtask
+      React.useEffect(() => {
+        Promise.resolve('red').then((c) => setColor(c));
```

Comment:
Pls add // eslint-ignore-next with a brief comment to fix lint

Reply:
Done ✅

Reply:
@jsnajdr Huge thanks for submitting both microtask-related PRs! This is complex subject, so I plan to reserve some time in the upocoming week to wrap my head around these.